### PR TITLE
Add dual-auth for Promxy at the root of the prometheus domain

### PR DESCRIPTION
**Pull request by mig4**
Restore OAuth2 authentication as this is much more user-friendly, and
add `satisfy any` directive which allows access if at least one auth
module (OAuth2/Basic) allows it (by default all must allow it).

Towards: https://github.com/giantswarm/giantswarm/issues/12864

### CODE REVIEW

File: helm/promxy-app/templates/ingress.yaml [hunk 1]
```diff
@@ -5,9 +5,19 @@ metadata:
...
+    # OAuth2 SSO access for humans
+    nginx.ingress.kubernetes.io/auth-signin: https://$host/oauth2/start?rd=$escaped_request_uri
+    nginx.ingress.kubernetes.io/auth-url: https://$host/oauth2/auth
+    # basic-auth for global Promxy to be able to access all installation-level
+    # Promxies
     nginx.ingress.kubernetes.io/auth-type: basic
     nginx.ingress.kubernetes.io/auth-secret: {{ include "resource.default.name" . }}
     nginx.ingress.kubernetes.io/auth-realm: 'Authentication Required'
+    # allow when any of the auth mechanisms succeeds
+    nginx.ingress.kubernetes.io/satisfy: any
```
Comment (QuentinBisson) [inline helm/promxy-app/templates/ingress.yaml:17]: Oh, this is really interesting

**Review by QuentinBisson — APPROVED** [review]

**Comment by mig4** [comment]
I've rebased this and simplified it a lot, down to one addition 😄 
In case someone wants to give it another glance.

**Comment by mig4** [comment]
BTW, tested this on _ghost_ and it works OK.

**Comment by mig4** [comment]
Going with this.

**Pull request merged**